arroyo/processing/processor.py

```diff
@@ -17,13 +17,48 @@
 
 logger = logging.getLogger(__name__)
 
-LOG_THRESHOLD_TIME = 10  # In seconds
+LOG_THRESHOLD_TIME = 5  # In seconds
 
 
 class InvalidStateError(RuntimeError):
     pass
 
 
+class MetricsBuffer:
+    def __init__(self) -> None:
+        self.__metrics = get_metrics()
+        self.__metrics_frequency_sec = 1.0
+        self.__reset()
+
+    def add_poll_time(self, duration: float) -> None:
+        self.__consumer_poll_time += duration
+        self.__throttled_record()
+
+    def add_processing_time(self, duration: float) -> None:
+        self.__processing_time += duration
+        self.__throttled_record()
+
+    def add_pause_time(self, duration: float) -> None:
+        self.__paused_time += duration
+        self.__throttled_record()
+
+    def flush(self) -> None:
+        self.__metrics.timing("arroyo.consumer.poll.time", self.__consumer_poll_time)
+        self.__metrics.timing("arroyo.consumer.processing.time", self.__processing_time)
+        self.__metrics.timing("arroyo.consumer.paused.time", self.__paused_time)
```

Comment:
Just wanted to point out the metric naming convention being introduced here. I decided to prefix with `arroyo` as I think it keep things much better organized than they are now and avoids potential naming conflicts.

Reply:
I think I am ok with that, just bear in mind that the client provides a metric wrapper object to begin with that generally has a prefix already